File: adventjs/2025/24/solution.py
def is_trees_synchronized(tree1, tree2):
  root_value = tree1.get('value', '') if tree1 else ''
  
  if not tree1 and not tree2:
    return [True, root_value]
  if not tree1 or not tree2:
    return [False, root_value]
  if tree1.get('value') != tree2.get('value'):
    return [False, root_value]
  
  left_sync, _ = is_trees_synchronized(tree1.get('left'), tree2.get('right'))
  right_sync, _ = is_trees_synchronized(tree1.get('right'), tree2.get('left'))
  
  return [left_sync and right_sync, root_value]
  

File: adventjs/2025/24/solution.js
/**
 * @param {object} tree1 - The first binary tree.
 * @param {object} tree2 - The second binary tree.
 * @returns {[boolean, string]}
 */
function isTreesSynchronized(tree1, tree2) {
  if (!tree1 && !tree2) return [true, ''];
  if (!tree1 || !tree2) return [false, tree1?.value || tree2?.value || ''];
  if (tree1.value !== tree2.value) return [false, tree1.value];
  
  const [leftSync] = isTreesSynchronized(tree1.left, tree2.right);
  const [rightSync] = isTreesSynchronized(tree1.right, tree2.left);
  
  return [leftSync && rightSync, tree1.value];
}
  